governance/program/src/addins/voter_weight.rs

```diff
@@ -0,0 +1,110 @@
+//! VoterWeight Addin interface
+
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use solana_program::{
+    account_info::AccountInfo,
+    clock::{Clock, Slot},
+    program_error::ProgramError,
+    program_pack::IsInitialized,
+    pubkey::Pubkey,
+    sysvar::Sysvar,
+};
+
+use crate::{
+    error::GovernanceError,
+    state::token_owner_record::TokenOwnerRecord,
+    tools::account::{get_account_data, AccountMaxSize},
+};
+
+/// VoterWeight account type
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub enum VoterWeightAccountType {
+    /// Default uninitialized account state
+    Uninitialized,
+
+    /// Voter Weight Record
+    VoterWeightRecord,
+}
+
+/// VoterWeight Record account
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub struct VoterWeightRecord {
+    /// VoterWeightRecord account type
+    pub account_type: VoterWeightAccountType,
+
+    /// The Realm the VoterWeightRecord belongs to
+    pub realm: Pubkey,
+
+    /// Governing Token Mint the VoterWeightRecord is associated with
+    /// Note: The addin can take deposits of any tokens and is not restricted to the community or council tokens only
+    // The mint here is to link the record to either community or council mint of the realm
+    pub governing_token_mint: Pubkey,
+
+    /// The owner of the governing token and voter
+    pub governing_token_owner: Pubkey,
+
+    /// Voter's weight
+    pub voter_weight: u64,
```

Comment:
Without the addin, is this number normally the same as number of "community tokens" one votes with?

Reply:
Yes. It’s the number of deposited tokens